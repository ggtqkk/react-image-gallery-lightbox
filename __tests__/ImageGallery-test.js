/**
 * ImageGallery tests for JEST
 *
 * Copyright Â© Oleg Nosov 2016
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */
import { mount } from 'enzyme';
import React from 'react';
import ImageGallery from '../src/image-gallery';

const createGallery = props =>
  mount(
    <ImageGallery
      { ...props }
    />);


describe('ImageGallery', () => {
  const galleryProps = {
    count: 3,
    path: '',
  };

  it('checks if count is 3 then 4 images will be on page (3 thumbnails and main)', () => {
    const renderedGallery = createGallery(galleryProps);
    expect(renderedGallery.find('img').length).toBe(4);
  });

  it('checks if count is 1 then 1 image will be on page (main only)', () => {
    const renderedGallery = createGallery({ ...galleryProps, count: 1 });
    expect(renderedGallery.find('img').length).toBe(1);
  });

  it('takes snapshot', () => {
    const renderedGallery = createGallery(galleryProps);
    expect(renderedGallery.html()).toMatchSnapshot();
  });
});
