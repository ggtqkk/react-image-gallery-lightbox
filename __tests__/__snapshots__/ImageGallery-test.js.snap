exports[`ImageGallery takes snapshot 1`] = `
"<div><div class=\"row\"><div class=\"col-xs-12 col-sm-6 col-md-5 col-lg-6\"><div><div class=\"text-xs-center m-b-1\"><img role=\"button\" class=\"img-fluid m-x-auto d-block \" src=\"1-483x321.jpeg\"></div><div class=\"row\"><div class=\"col-xs-4
            col-md-4
            text-xs-center\"><img role=\"button\" src=\"1-102x68.jpeg\" class=\"img-fluid img-thumbnail m-b-1 image-gallery-selected\"></div><div class=\"col-xs-4
            col-md-4
            text-xs-center\"><img role=\"button\" src=\"2-102x68.jpeg\" class=\"img-fluid img-thumbnail m-b-1\"></div><div class=\"col-xs-4
            col-md-4
            text-xs-center\"><img role=\"button\" src=\"3-102x68.jpeg\" class=\"img-fluid img-thumbnail m-b-1\"></div></div></div></div><div class=\"col-xs-12 col-sm-6 col-md-7 col-lg-6\"></div></div><!-- react-empty: 15 --></div>"
`;
